src/pages/CollectionSingleViewPage.tsx

```diff
@@ -359,6 +400,144 @@ const TabPanel = ({ children, value, index, ...other }: TabPanelProps) => {
   );
 };
 
+type MutationTableTabContentProps = {
+  mutationType: string;
+  locationSelector: LocationSelector;
+  variants: { query: VariantSelector; name: string; description: string }[];
+  dateRangeSelector: DateRangeSelector;
+  mode: 'Single' | 'CompareToBaseline';
+  baselineVariant?: VariantSelector;
+};
+
+const MutationTableTabContent = ({
+  mutationType,
+  variants,
+  locationSelector,
+  dateRangeSelector,
+  mode,
+  baselineVariant,
+}: MutationTableTabContentProps) => {
+  const mutations = useQuery(
+    async signal => {
+      if (!variants) {
+        return undefined;
+      }
+      const data: MutationObject[] = [];
+      const dataMutations = await Promise.allSettled(
+        variants.map(variant => {
+          const selector: LapisSelector = addDefaultHostAndQc({
+            location: locationSelector,
+            variant: variant.query,
+            samplingStrategy: SamplingStrategy.AllSamples,
+            dateRange: dateRangeSelector,
+          });
+
+          return Promise.all([
+            fetchSamplesCount(selector, signal),
+            MutationProportionData.fromApi(selector, 'aa', signal),
+            MutationProportionData.fromApi(selector, 'nuc', signal),
+          ]).then(async ([variantCount, aaMutationDataset, nucMutationDataset]) => {
+            data.push({
+              ...variant,
+              variantCout: variantCount,
+              aaMutations: aaMutationDataset.payload,
+              nucMutations: nucMutationDataset.payload,
+            });
+          });
+        })
+      );
+
+      return data;
+    },
+    [variants, locationSelector, dateRangeSelector]
+  );
+
+  const baselineMutations = useQuery(
+    async signal => {
+      const data: BaselineData = {
+        aaMutations: [],
+        nucMutations: [],
+      };
+
+      if (mode === 'CompareToBaseline') {
+        const selector: LapisSelector = addDefaultHostAndQc({
+          location: locationSelector,
+          variant: baselineVariant,
+          samplingStrategy: SamplingStrategy.AllSamples,
+          dateRange: dateRangeSelector,
+        });
+
+        const baselineData = Promise.all([
+          fetchSamplesCount(selector, signal),
+          MutationProportionData.fromApi(selector, 'aa', signal),
+          MutationProportionData.fromApi(selector, 'nuc', signal),
+        ]).then(async ([variantCount, aaMutationDataset, nucMutationDataset]) => {
+          data.aaMutations = aaMutationDataset.payload;
+          data.nucMutations = nucMutationDataset.payload;
+        });
+
+        console.log('baseline', data);
+
+        return data;
+      }
+    },
+    [baselineVariant, mode, locationSelector, dateRangeSelector]
+  );
+
+  // #####################################
```

Comment:
How about extracting some of those hook usages in custom hooks? That would make the function more readable and we don't need this visual separator in the code.